### PR TITLE
Reworked the login dialog to avoid redundant label/placeholder #3459

### PR DESCRIPTION
**Pull request by radinamatic**
Not exactly as the example @Antrikshy posted, but close.
A11y tested with NVDA on FF.

### CODE REVIEW

**Comment by Antrikshy** [comment]
@radinamatic Screenshots?

**Comment by radinamatic** [comment]
You're reading my mind...? Just was making one... :) 

login-dialog

File: kalite/distributed/hbtemplates/user/login.handlebars [hunk 1]
```diff
@@ -4,9 +4,10 @@
             <form class="form-horizontal">
             {{#if facilities}}
             <div class="form-group form-group-lg" id="id_facility-container">
-              <label class="col-sm-3 control-label h3" for="id_facility" id="id_facility-addon">{{_ "Facility" }}</label>
-              <div class="col-sm-9">
-                <select class="form-control input-lg" id="id_facility" aria-describedby="id_facility-addon">
+              <label class="sr-only control-label h3" for="id_facility">{{_ "Facility" }}</label>
```
Comment (rtibbles) [inline kalite/distributed/hbtemplates/user/login.handlebars]: Any reason for removing the id tag here?

File: kalite/distributed/static/css/distributed/bootstrap-overrides.css [hunk 2]
```diff
@@ -290,6 +290,11 @@ form .input-group {
     background-color: #40775F !important;
 }
 
+.login-btn, btn-block {
+    display: block;
+    width: 100%;
```
Comment (rtibbles) [inline kalite/distributed/static/css/distributed/bootstrap-overrides.css]: Can you set SIMPLIFIED_LOGIN=True in your local_settings.py and see how this width setting interacts with the additional button that appears with this setting?

File: kalite/distributed/static/css/distributed/bootstrap-overrides.css [hunk 3]
```diff
@@ -455,6 +460,18 @@ footer {
     width: 82px;
 }
 
+/* ===================================
+    Modal Dialog Login
+=================================== */
+
+.modal-dialog {
```
Comment (rtibbles) [inline kalite/distributed/static/css/distributed/bootstrap-overrides.css]: Why do we want to override all bootstrap modal dialog boxes? Is there an issue with the way that bootstrap handles the sizing?

File: kalite/distributed/static/css/distributed/bootstrap-overrides.css [hunk 4]
```diff
@@ -455,6 +460,18 @@ footer {
...
+    width: 400px;
+}
+
+div.panel-body form.form-horizontal .form-group {
```
Comment (rtibbles) [inline kalite/distributed/static/css/distributed/bootstrap-overrides.css]: Again, I would rather not do such non-specific selectors there. Can you make this more specific to the login box where we want this behaviour to occur (and hence move it out of bootstrap-overrides also).

**Comment by rtibbles** [comment]
A few concerns with over generalized CSS selectors (using a hammer to crack a nut), also the login button width might be a problem in the situation mentioned above.

Lastly, error handling will now be somewhat subtle. There is a popover that will show the error message, but the highlighting of the input box is very subtle on bootstrap defaults. Please have a look at the behaviour there with the addition of the 'has-error' class to the containing div of the form elements.

File: kalite/distributed/static/css/distributed/bootstrap-overrides.css (hunk 4 above)
Comment (radinamatic) [inline kalite/distributed/static/css/distributed/bootstrap-overrides.css]: Ok, will work on more-specific-selector solution.
Question: if the changes are to be moved **from** bootstrap-overrides, where do we move them to?

Comment (rtibbles) [inline kalite/distributed/static/css/distributed/bootstrap-overrides.css]: Probably into khan-lite.css would be best.

File: kalite/distributed/static/css/distributed/bootstrap-overrides.css (hunk 3 above)
Comment (radinamatic) [inline kalite/distributed/static/css/distributed/bootstrap-overrides.css]: No reason I can think of right now, I will eliminate this.

File: kalite/distributed/static/css/distributed/bootstrap-overrides.css (hunk 2 above)
Comment (radinamatic) [inline kalite/distributed/static/css/distributed/bootstrap-overrides.css]: This is how it looks with simplified login:

simplified-login-dialog

File: kalite/distributed/hbtemplates/user/login.handlebars (hunk 1 above)
Comment (radinamatic) [inline kalite/distributed/hbtemplates/user/login.handlebars]: Must be the Git gremlin, I'll put it back... :)

File: kalite/distributed/static/css/distributed/bootstrap-overrides.css (hunk 2 above)
Comment (rtibbles) [inline kalite/distributed/static/css/distributed/bootstrap-overrides.css]: Odd, the password box should be hidden, and the login button should be disabled until a username is typed in.

Comment (rtibbles) [inline kalite/distributed/static/css/distributed/bootstrap-overrides.css]: Also, could you set the lower button to be centrally aligned? I think it is set to float: right currently.

Comment (rtibbles) [inline kalite/distributed/static/css/distributed/bootstrap-overrides.css]: Or is this after pressing the 'Admin/Coach' button?

Comment (radinamatic) [inline kalite/distributed/static/css/distributed/bootstrap-overrides.css]: Check the latest commit:

a) Login dialog loaded
simplified-login-dialog2

b) Wrong username no password
simplified-login-dialog3

c) Wrong username & password
simplified-login-dialog4

One thing I noticed is that the error message tooltip stays on username even when it is correctly inputed for the given facility, and the password is the incorrect one. I'm guessing lines 219-220 in /facility/forms.py?

**Comment by radinamatic** [comment]
I possibly overdid a bit the "un-subtling" of the input fields with 'has-error', but that's easily adjustable.

Will have to work on accessibility of those popover tooltips though,  they only get read if I put aria-live="assertive" which is another hammer... :)

File: kalite/distributed/static/css/distributed/bootstrap-overrides.css (hunk 2 above)
Comment (rtibbles) [inline kalite/distributed/static/css/distributed/bootstrap-overrides.css]: No - this validation code is either happening in facility/api_resources.py or in distributed/static/js/user/views.js

Incorrect popover behaviour is probably happening in the latter.

**Comment by rtibbles** [comment]
Sorry for letting this get stale, @radinamatic - could you rebase, and maybe tone the has-error back down a little and then I'll merge this.

**Comment by radinamatic** [comment]
Just seeing your last comment here @rtibbles, I'm not totally satisfied with the manner SRs deal with the tooltips here... As soon as I find the acceptable solution, I'll rebase, adjust the CSS and commit again.

**Pull request closed without merging**